### PR TITLE
Add StatusString into JwtAuthn filter's rc_detail

### PR DESCRIPTION
**Pull request by TAOXUY**
Commit Message: Add the status string into the jwtAuthn filter's response code detail. The change is simply
`jwt_authn_access_denied` -> `jwt_authn_access_denied{#STATUS_STRING}`, ex. `jwt_authn_access_denied{Audiences in Jwt are not allowed}`. 

It provides more information why requests are rejected in the access log to the service provider, which is same as what the downstream see right now. It shouldn't break anything unless some automation depends on the static `jwt_authn_access_denied` %RESPONSE_CODE_DETAIL%, which is very unlikely.

Risk Level:low

Testing:included in the integration test

### CODE REVIEW

**Comment by TAOXUY** [comment]
@qiwzhang @nareddyt

**Review by qiwzhang — APPROVED** [review]

**Comment by qiwzhang** [comment]
Please use "tools/spelling/check_spelling.sh" to fix the format

**Comment by TAOXUY** [comment]
> Please use "tools/spelling/check_spelling.sh" to fix the format

Done.

**Comment by TAOXUY** [comment]
PTAL

File: source/extensions/filters/http/jwt_authn/filter.cc [hunk 1]
```diff
@@ -27,14 +27,10 @@ bool isCorsPreflightRequest(const Http::RequestHeaderMap& headers) {
          !headers.getInlineValue(access_control_request_method_handle.handle()).empty();
 }
 
+// The prefix used in the response code detail sent from jwt authn filter.
+constexpr char kRcDetailJwtAuthnPrefix[] = "jwt_authn_access_denied";
```
Comment (mattklein123) [inline source/extensions/filters/http/jwt_authn/filter.cc:31]: nit: make this absl::string_view to avoid any strlen lookup.

**Review by mattklein123 — CHANGES_REQUESTED** [review]
LGTM with small comment, thanks!

/wait

File: source/extensions/filters/http/jwt_authn/filter.cc (hunk 1 above)
Comment (TAOXUY) [reply source/extensions/filters/http/jwt_authn/filter.cc]: Done. Thanks!

**Review by mattklein123 — APPROVED** [review]
Thanks!

**Pull request merged**